lib/improver/tests/ensemble_copula_coupling/ensemble_copula_coupling/test_FromMeanAndVariance.py

```diff
@@ -0,0 +1,127 @@
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Unit tests for FromMeanAndVariance
+
+"""
+import unittest
+
+import numpy as np
+from scipy import stats
+from iris.tests import IrisTest
+
+from improver.ensemble_copula_coupling.ensemble_copula_coupling import (
+    FromMeanAndVariance as Plugin)
+
+
+class Test__init__(IrisTest):
+
+    """Test the __init__ method."""
+
+    def test_valid_distribution(self):
+        """Test for a valid distribution."""
+        plugin = Plugin(distribution="norm")
+        self.assertEqual(plugin.distribution, stats.norm)
+        self.assertEqual(plugin.shape_parameters, [])
+
+    def test_valid_distribution_with_shape_parameters(self):
+        """Test for a valid distribution with shape parameters."""
+        plugin = Plugin(distribution="truncnorm", shape_parameters=[0, np.inf])
+        self.assertEqual(plugin.distribution, stats.truncnorm)
+        self.assertEqual(plugin.shape_parameters, [0, np.inf])
+
+    def test_invalid_distribution(self):
+        """Test for an invalid distribution."""
+        msg = "The distribution requested"
+        with self.assertRaisesRegex(AttributeError, msg):
+            Plugin(distribution="elephant")
+
+
+class Test__repr__(IrisTest):
+
+    """Test string representation of plugin."""
+
+    def test_basic(self):
+        """Test string representation"""
+        expected_string = ("<FromMeanAndVariance: "
+                           "distribution: norm; shape_parameters: []>")
+        result = str(Plugin())
+        self.assertEqual(result, expected_string)
+
+
+class Test__rescale_shape_parameters(IrisTest):
+
+    """Test the _rescale_shape_parameters"""
+
+    def setUp(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method name "setUp" doesn't conform to '((test_[a-zA-Z0-9_]+)|([a-z_][a-z0-9_]{2,30}))$' pattern](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4540588)